commit 714b04d7eb514452aa0ae63b7fd6ee04c4ad888f (refs/changes/45/2345/1)
Author: Tim Kordas <tim.kordas@joyent.com>
Date:   2017-08-08T13:31:37-07:00 (2 years, 2 months ago)
    
    MANTA-3187: PUTs now log new objectId and previous if available

diff --git a/lib/audit.js b/lib/audit.js
index d352464..e98b1a1 100644
--- a/lib/audit.js
+++ b/lib/audit.js
@@ -227,8 +227,19 @@ function auditLogger(options) {
             secure: req.secure
         };
 
+        /*
+         * The req.metadata holds the original proposed objectId. If our
+         * request has an updated objectId, then it will be different, and
+         * be in req.objectId prefer to report that, and to report the proposal
+         * as prevObjectId.
+         */
         if (req.metadata && req.metadata.objectId) {
-            obj.objectId = req.metadata.objectId;
+            if (req.objectId && req.objectId !== req.metadata.objectId) {
+                obj.prevObjectId = req.metadata.objectId;
+                obj.objectId = req.objectId;
+            } else {
+                obj.objectId = req.metadata.objectId;
+            }
         }
         obj.sharksContacted = req.sharksContacted;
         obj.entryShard = req.entryShard;
